AltStore/Operations/ResignAppOperation.swift

```diff
@@ -113,15 +113,24 @@ private extension ResignAppOperation
             
             infoDictionary[kCFBundleIdentifierKey as String] = profile.bundleIdentifier
             infoDictionary[Bundle.Info.altBundleID] = identifier
-            
+
             for (key, value) in additionalInfoDictionaryValues
             {
                 infoDictionary[key] = value
             }
-            
+
             if let appGroups = profile.entitlements[.appGroups] as? [String]
             {
                 infoDictionary[Bundle.Info.appGroups] = appGroups
+
+                // To keep file providers working, remap the NSExtensionFileProviderDocumentGroup, if there is one.
+                if var extensionInfo = infoDictionary["NSExtension"] as? [String: Any],
+                    let appGroup = extensionInfo["NSExtensionFileProviderDocumentGroup"] as? String,
+                    let localAppGroup = appGroups.filter({ $0.starts(with: appGroup + ".") }).min(by: { $0.count < $1.count })
```

Comment:
Should check whether local app group contains expected app group, rather than starts with it.

```suggestion
                    let localAppGroup = appGroups.filter({ $0.contains(appGroup) }).min(by: { $0.count < $1.count })
```